onnxruntime/core/session/default_cpu_allocator_c_api.cc

```diff
@@ -23,10 +23,33 @@ struct OrtDefaultAllocator : OrtAllocatorImpl {
   ~OrtDefaultAllocator() override { OrtReleaseAllocatorInfo(cpuAllocatorInfo); }
 
   void* Alloc(size_t size) {
-    return ::malloc(size);
+    if (size == 0)
+      return nullptr;
+    //default align to 64;
+    void* p;
+#if defined(_WIN32) && !defined(_WIN64)
+  size_t alignment = 32;
+#else  
+  size_t alignment = 64;
+#endif
+#if _MSC_VER
```

Comment:
Some comments explaining the rationale behind these changes would be really helpful. Not clear why 32 is chosen on both win32 and win64 and why that's the best choice.

Reply:
Also investigate if MlasGetPreferredBufferAlignment() might be a better choice here. For systems with AVX-512, this returns 64 and for everybody else it returns 32. This is used in core\framework\allocator.cc.

Reply:
Good point, will reuse the function in core/framework/allocator.cc

---
In reply to: [312588800](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312588800) [](ancestors = 312588800)

Reply:
Will share the code from core\framework\allocator.cc

---
In reply to: [312421267](https://github.com/microsoft/onnxruntime/pull/881#discussion_r312421267) [](ancestors = 312421267)